### PR TITLE
Issue #1571: Move all dtd schemas to sourceforge site

### PR DESCRIPTION
**Pull request by MEZk**
Issue #1571

@romani 
Please, upload all dtd schemas to SourceForge and I will restart CIs build to fix failing UTs. I'll make a separate PR or commit to use http://www.mojohaus.org/xml-maven-plugin/examples/catalog.html .

### CODE REVIEW

File: src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!ELEMENT checkstyle-packages (package)*>
+
+<!ELEMENT package (package)*>
+<!ATTLIST package name NMTOKEN #REQUIRED>
```
Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd]: please recheck why Checkstyle is not catching this as violation. We enforce all files to have newline at the end

File: src/main/resources/com/puppycrawl/tools/checkstyle/configuration_1_4.dtd [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/configuration_1_4.dtd]: what is changed from 1_3 ?
Do we really need to bump version ? it is kind of strange that we change url in example - do not bump version.
and in the same time introduce 1_4.

We just need change files and copy them to new location.
Change all our XML files to point new location.
If user have old xml config that reference to old location of DTD - it should continue to work.
Files that we are packing and hosting should use new location.

Comment (MEZk) [inline src/main/resources/com/puppycrawl/tools/checkstyle/configuration_1_4.dtd]: @romani 
From your [comment](https://github.com/checkstyle/checkstyle/issues/1571#issuecomment-257854819): 

> all schemes are stored in git, all we need is to change them as described , bump their versions (for all except for schema that is changed at #2999)

File: src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd (hunk 1 above)
Comment (MEZk) [inline src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd]: @romani 
> We enforce all files to have newline at the end

No, we do not. See https://github.com/MEZk/checkstyle/blob/52d1237a32dd1ddd6ea4697643534f59d8bebb64/config/checkstyle_checks.xml#L21-L21 .

[The documentation](https://github.com/MEZk/checkstyle/blob/131723cab4ac2fd0953b877515a59399c81de002/src/xdocs/config_misc.xml#L1191-L1191) of NewLineAtEndOfFile is not correct. If we do not set fileExtensions property of AbstractFileSetCheck (NewLineAtEndOfFile extends AbstractFileSetCheck) Checker uses its own property fileExtensions.

**Comment by romani** [comment]
please rebase on our latest master to fix werker. TC failure due to license could be ignored, I reported and issue to their support.

**Comment by MEZk** [comment]
@romani 
Done.

File: src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd (hunk 1 above)
Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd]: fixed in  794637e commit.

Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd]: xdoc is fixed at a719c39 commit.

File: src/main/resources/com/puppycrawl/tools/checkstyle/configuration_1_4.dtd (hunk 2 above)
Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/configuration_1_4.dtd]: > So what do we need to change? Only comments in dtd schemes?

if we change only comment - version should not be bumped.

> Do you plan to upload 1_0 - 1_X schemes to sourceforge too?

yes, all old versions on dtd should be accessible, users still works on very old versions of checkstyle.

**Comment by romani** [comment]
@MEZk ,

I copied current version of dtd files to sourceforge
```
$ scp *.dtd  romanivanov,checkstyle@shell.sourceforge.net:/home/project-web/checkstyle/htdocs/dtds/
configuration_1_0.dtd                                                                                                         100%  441     0.4KB/s   00:00    
configuration_1_1.dtd                                                                                                         100% 1020     1.0KB/s   00:00    
configuration_1_2.dtd                                                                                                         100% 1047     1.0KB/s   00:00    
configuration_1_3.dtd                                                                                                         100% 1629     1.6KB/s   00:00    
packages_1_0.dtd                                                                                                              100%  155     0.2KB/s   00:00    
suppressions_1_0.dtd                                                                                                          100%  488     0.5KB/s   00:00    
suppressions_1_1.dtd                                                                                                          100%  524     0.5KB/s   00:00

$ scp *.dtd  romanivanov,checkstyle@shell.sourceforge.net:/home/project-web/checkstyle/htdocs/dtds/
import_control_1_0.dtd                                                                                                        100% 1928     1.9KB/s   00:00    
import_control_1_1.dtd                                                                                                        100% 2046     2.0KB/s   00:00    
import_control_1_2.dtd                                                                                                        100% 2220     2.2KB/s   00:00 
```

They are accessible from web.
So you can update all source files to use new location of dtd files, CIs should pass.

Please rebase on latest our master to resolve other CIs problems.

File: config/checkstyle_checks.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <!DOCTYPE module PUBLIC
-          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
-          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+          "-//Puppy Crawl//DTD Check Configuration 1.4//EN"
+          "http://checkstyle.sourceforge.net/dtds/configuration_1_4.dtd">
```
Comment (romani) [inline config/checkstyle_checks.xml]: as we do not introduce any new functionality in DTD , lets keep it version.
Please use http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd , it should work.

Please do the same for others.

Comment (MEZk) [inline config/checkstyle_checks.xml]: Fixed

**Comment by MEZk** [comment]
@romani 
> So you can update all source files to use new location of dtd files, CIs should pass.
Please rebase on latest our master to resolve other CIs problems.

Done.

**Comment by romani** [comment]
thanks a lot, but CIs red 
```
[checkstyle] [ERROR] /home/travis/build/checkstyle/checkstyle/config/catalog.xml:0: File does not end with a newline. [NewlineAtEndOfFile]
```

File: pom.xml [hunk 4]
```diff
@@ -957,6 +957,9 @@
               <systemId>src/main/resources/com/puppycrawl/tools/checkstyle/configuration_1_3.dtd</systemId>
             </validationSet>
           </validationSets>
+          <catalogs>
+            <catalog>config/catalog.xml</catalog>
+          </catalogs>
```
Comment (romani) [inline pom.xml]: @MEZk , why this become required ? or ..... 
please explain .

**Comment by MEZk** [comment]
@romani 
> thanks a lot, but CIs red

Fixed.

File: pom.xml (hunk 4 above)
Comment (MEZk) [inline pom.xml]: @romani 
It is the config for Catalog Files feature of maven xml plugin.
See http://www.mojohaus.org/xml-maven-plugin/examples/catalog.html

Comment (romani) [inline pom.xml]: I know this, just curious why to put it right now

**Comment by romani** [comment]
@rnveach , please review

File: pom.xml (hunk 4 above)
Comment (MEZk) [inline pom.xml]: @romani 
I made it as a separate commit and checked that this approach really works without Internet connection.

**Pull request merged**